### DISCUSSION TITLE
no plot in jupyter notebook with pyvista

### DISCUSSION THREAD

**Discussion by emi0000** [pyvista/pyvista#3207] [Q&A]
I'using jupyter notebook with ubuntu 22.4 and installed ipygany (0.5.0), pythreejs (2.3.0) and ipyvtklink (0.2.2). However I do not get any plot

      import pyvista as pv
      pv.set_jupyter_backend('ipygany')

      poi=np.array([[0.5,1,0],[2,1,0],[3,1.5,0],[3.5,2.5,0],[2.2,2,0],[1,2.2,0]])
      tri=np.array([[3,0,1,5],[3,1,4,5],[3,1,2,4],[3,2,3,4]])  
      surf = pv.PolyData(poi,tri)
      surf.plot(show_edges=True) 

By replacing 'iygany' with 'static' I'm getting a plot. The above code works fine in an IDE omitting the backend line. It does not work with the pvista example either.

      import pyvista as pv
      from pyvista import examples

      # download an example and display it using physically based rendering.
      mesh = examples.download_lucy()
      mesh.plot(color='lightgrey', pbr=True, metallic=0.2 , jupyter_backend='pythreejs')

I also tried
jupyter nbextension enable --py --sys-prefix ipygany
however here I'm getting the message
ModuleNotFoundError: No module named 'ipygany'
but ipygany is installed when checking with pip list

**Comment by akaszynski** [comment]
`ipygany` hasn't been committed for over a year. Have you tried another backend like `panel` or `pythreejs`?

**Reply by emi0000** [reply]
I've tried ipygany, pythreejs and ipyvtklink. For all three backends I did not get any output. 

I just tried panel, that works, thanks.

Do you know if a working command
jupyter nbextension enable --py --sys-prefix ipygany
may enable also other backends? I saw that command somewhere but was not able to make it work.

**Comment by DanielJamesEvans** [comment]
I think downgrading ipywidgets might solve your issue.  This is discussed in https://github.com/pyvista/pyvista/issues/3274 and https://github.com/pyvista/pyvista/issues/3282.  The command `pip install ipywidgets==7.7.1` worked for me.

**Reply by emi0000** [reply]
yes, it works for me too. I tested with ipygany, pythreejs and ipyvtklink, all of them come up with an output now. Thanks a lot.
